### ISSUE TITLE
Support for `multiValueHeaders` in response

### ISSUE THREAD

**Issue by andybee** [architect/architect#764]
**Is your feature request related to a problem? Please describe.**
I'm trying to set multiple cookies in a response, which I can't do unless I can send 2x `Set-Cookie` headers.

**Describe the solution you'd like**
I can see two ways of implementing this, either adding support for the `headers` object in a response to have either arrays of strings as the value for each key and, when it's an array, handling that as multiple headers:

```
exports.handler = function foo() {
  return {
    statusCode: 200,
    headers: {
      'Set-Cookie': ['a=1', 'b=2'],
    },
  };
}
```

Alternatively, for cleanliness (but at the risk of complicating the response object) adding an explicit `multiValueHeaders` property.

```
exports.handler = function foo() {
  return {
    statusCode: 200,
    multiValueHeaders: {
      'Set-Cookie': ['a=1', 'b=2'],
    },
  };
}
```

**Comment by ryanblock** [comment]
That could be neat, but unfortunately you are not responding to Architect here, you're responding to API Gateway, so the shape of your response needs to meet their API.

And of course because this is AWS, for the life of me, this is the best documentation I can find that's available that explains the shape of what you can return: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-integration-settings-integration-response.html

tl;dr you can't do dynamic / parsed stuff, it's all strings, ints, and bools.

This could become a feature of @architect/functions, however. PRs welcome!

**Comment by andybee** [comment]
Aha, I didn't realise there was no abstraction in the middle at that point.

In which case it feels like a "bug" (used in the nicest sense of the word) in Sandbox that it rejects `multiValueHeaders` as invalid, as API Gateway will accept it. And here's my best shot at finding a seemingly buried, not necessarily authoritative [snippet from AWS to demonstrate that support](https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-lambda-proxy-integrations.html#api-gateway-simple-proxy-for-lambda-output-format)(!)

I'll see if I can work out how to add support for that response parameter in Sandbox.

**Comment by ryanblock** [comment]
TIL! Yeah if so we should def support it in sandbox. Where is this documented? Would love to see if sears missing any others.

**Comment by andybee** [comment]
My best attempt at ever finding an authority on this is following the trail that API Gateway should be defining what it expects the Lambda to respond. So, diving in to the API Gateway docs I find the above one I linked to and, similarly, this bit:

https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html#http-api-develop-integrations-lambda.response

Luckily, at the time of writing, they agree with each other.

**Comment by andybee** [comment]
Actually, worth drawing attention to the last document I linked to, what seems to be the most authoritative guide on both request and response formats and the differences between v1 and v2. This might have some implications for Architect in the future as v2 request format differs quite a bit with regard to `multiX` properties being merged.

It would probably make sense for Architect to update handling to v2 where relevant and explicitly set `payloadFormatVersion` to 2.0?

**Comment by ryanblock** [comment]
Thanks for schooling us on this! I shared this to other folks and no one else realized `multiValueHeaders` was a valid response payload, which is kind of amazing to me.

Also correct me if I'm wrong, but `multiValueHeaders` isn't valid in response format 2.0?

I know @brianleroux has been doing some work on the new format in preparation for HTTP APIs to mature. We haven't mainlined that to Arc yet though because it's still missing a lot of service integrations, but you can test it out here: https://github.com/architect/macro-http-api

**Comment by andybee** [comment]
Great news on eventually flipping over to HTTP API Gateway (seriously, AWS, what kind of name is that for a product?!), think that has some real benefits. I'll be sure to check out the macro shortly.

I think you're right on it being dropped from 2.0. Looking at the changes to the request format, I wonder if they'll break out:
```
{ header: 'foo,bar' }
```
to:
```
Header: foo
Header: bar
```
?

**Comment by ryanblock** [comment]
Will be released in `1.9.6`, thanks again @andybee, excellent job!
